### ISSUE TITLE
[MU4 Issue] Playback of note doesn't respect tempo change during note

### ISSUE THREAD

**Issue by Fermats-Fish** [musescore/MuseScore#12884]
**Describe the bug**
If the tempo changes during the playback of a note (e.g. the note is tied to the next bar where the tempo changes) then the duration the note is played for does not factor in the change of tempo.

**To Reproduce**
1. Create a new score for solo violin
2. Set the tempo of the first bar to 80 bpm, and the second 160
3. Add a note to the first bar and tie it to a note of the same pitch in the second bar.
4. Place another note of different pitch directly after the first note ends.
5. Play back the score, and notice that the two different pitches sound at the same time.

**Expected behavior**
The sound of the second note should begin just after the first note stops

**Screenshots**
<img width="354" alt="image" src="https://user-images.githubusercontent.com/3761567/184859919-427b3f49-6dab-4a31-8336-79cefbcdeeab.png">

**Platform information**
 - OS: macOS
 - Version: Alpha 2

**Comment by Tantacrul** [comment]
I believe this has been fixed.

**Comment by Fermats-Fish** [comment]
@Tantacrul I just retested this in the latest nightly build and it works correctly now.

**Comment by vpereverzev** [comment]
> @Tantacrul I just retested this in the latest nightly build and it works correctly now.

yes, I've fixed similar issue recently - https://github.com/musescore/MuseScore/issues/13325
